src/main/java/RacingGame.java

```diff
@@ -0,0 +1,40 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+class RacingGame {
+
+    private List<Car> cars = new ArrayList<>();
+    private List<Car> winnerCars = new ArrayList<>();
+
+    public void setCars(List<Car> cars) {
+        this.cars = cars;
+    }
+
+    public void gameStart(int rounds) {
+        Random random = new Random();
+        while (rounds-- > 0) {
+            moveCars(random);
+            System.out.println();
+        }
+    }
+
+    public void moveCars(Random random) {
+        for (Car car : cars) {
+            System.out.print(car.getName() + " : ");
+            int randomValue = random.nextInt(10);
+            car.move(randomValue);
+            System.out.print("-".repeat(car.getPosition()));
+            System.out.println();
+        }
+    }
+
+    public List<Car> getWinner() {
+        int maxPosition = cars.stream().mapToInt(Car::getPosition).max().orElse(0);
+        winnerCars = cars.stream().filter(car -> car.getPosition() == maxPosition).collect(Collectors.toList());
+        return winnerCars;
+    }
```

Comment:
```suggestion
    public List<Car> getWinner() {
        int maxPosition = cars.stream()
                              .mapToInt(Car::getPosition)
                              .max()
                              .orElse(0);
        
        return cars.stream()
                   .filter(car -> car.getPosition() == maxPosition)
                   .collect(Collectors.toList());
    }
```

이와 같이 수정하면 단계별로 표현되어서 더 읽기 쉬울 것 같아요 :)

Reply:
추가로 반환되는 `List<Car>`는 가변 객체인가요 불변 객체인가요?

Reply:
네 감사합니다! 단계별로 수정하였습니다!

현재 코드에서는 가변 객체가 반환되고 있습니다. 가변 리스트는 추가, 삭제, 수정하는 것이 가능하기에 데이터의 무결성이 깨질 위험이 있는 것 같습니다. 
예기치 못하게 우승자의 리스트가 바뀌는 상황을 방지하기 위해서 불변 리스트로 반환해주는 개선이 필요해 보입니다. 
가변 객체와 불변 객체를 생각해보며 리팩토링 진행하였습니다. 좋은 피드백 감사합니다!